assets/js/modules/adsense/components/common/AdSenseConnectCTA.js

```diff
@@ -0,0 +1,252 @@
+	createInterpolateElement,
+	useEffect,
+	useCallback,
+	useContext,
+	useState,
+} from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { MODULES_ADSENSE } from '../../datastore/constants';
+import { Grid, Row, Cell } from '../../../../material-components';
+import { ADSENSE_CTA_WIDGET_DISMISSED_ITEM_KEY } from '../../constants';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import { CORE_MODULES } from '../../../../googlesitekit/modules/datastore/constants';
+import { CORE_LOCATION } from '../../../../googlesitekit/datastore/location/constants';
+import { trackEvent } from '../../../../util';
+import Link from '../../../../components/Link';
+import Button from '../../../../components/Button';
+import Portal from '../../../../components/Portal';
+import Dialog from '../../../../components/Dialog';
+import AdSenseIcon from '../../../../../svg/adsense.svg';
+import ViewContextContext from '../../../../components/Root/ViewContextContext';
+const { useSelect, useDispatch } = Data;
+
+export default function AdSenseConnectCTA() {
+	const [ inProgress, setInProgress ] = useState( false );
+	const [ dialogActive, setDialogActive ] = useState( false );
+
+	const { dismissItem } = useDispatch( CORE_USER );
+	const { navigateTo } = useDispatch( CORE_LOCATION );
+	const { activateModule } = useDispatch( CORE_MODULES );
+	const { setInternalServerError } = useDispatch( CORE_SITE );
+
+	const viewContext = useContext( ViewContextContext );
+
+	const supportURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getGoogleSupportURL( {
+			path: '/adsense/answer/9724',
+		} )
+	);
+	const adminReauthURL = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getAdminReauthURL()
+	);
+	const adSenseModuleActive = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleActive( 'adsense' )
+	);
+	const adSenseModuleConnected = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleConnected( 'adsense' )
+	);
+
+	const handleConnect = useCallback( async () => {
+		const { response, error } = await activateModule( 'adsense' );
+
+		if ( error ) {
+			setInternalServerError( {
+				id: 'setup-module-error',
+				description: error.message,
+			} );
+			return null;
+		}
+
+		await trackEvent(
+			`${ viewContext }_widget-activation-cta`,
+			'activate_module',
+			'adsense'
+		);
+
+		navigateTo( response.moduleReauthURL );
+	}, [ activateModule, navigateTo, setInternalServerError, viewContext ] );
+
+	const handleCompleteSetup = useCallback(
+		() => navigateTo( adminReauthURL ),
+		[ adminReauthURL, navigateTo ]
+	);
+
+	const handleDismissModule = useCallback( () => {
+		setDialogActive( true );
+	}, [] );
+
+	const handleConfirmDialog = useCallback( async () => {
+		setInProgress( true );
+		await dismissItem( ADSENSE_CTA_WIDGET_DISMISSED_ITEM_KEY );
+		setInProgress( false );
```

Comment:
Instead of an extra state variable here, I think it would be a cleaner solution to use the `isFetchingDismissItem` selector that is automatically generated by using `createFetchStore`. If we wanted to make for a bit of a nicer API, we could also introduce an `isDismissingItem( slug )` selector which just wraps `isFetchingDismissItem`.

This way you wouldn't need `useState` for the `inProgress` variable here, it could instead be a single variable assignment like `const inProgress = useSelect( ( select ) => isDismissingItem( slug ) )`.

Reply:
Nice suggestion @felixarntz . Implemented by adding the `isDismissingItem` selector.